main.tf

```
@@ -178,7 +178,8 @@ resource "aws_codedeploy_deployment_group" "default" {
   # Note that you cannot have both ec_tag_filter and ec2_tag_set vars set!
   # See https://docs.aws.amazon.com/cli/latest/reference/deploy/create-deployment-group.html for details
   dynamic "ec2_tag_filter" {
-    for_each = length(var.ec2_tag_filter) > 0 ? [] : var.ec2_tag_filter
+    for_each = var.ec2_tag_filter
+
     content {
       key   = ec2_tag_filter.value["key"]
       type  = ec2_tag_filter.value["type"]
```

couldn't we use lookups here that result in null if the key is missing?

```suggestion
      type  = lookup(ec2_tag_filter.value, "type", null)
```

---

ah... yes, sure. I'll amend the PR in a moment. Thank you for this! This code is going to be pretty robust once we're done! :)